### PR TITLE
Allow to re allocate the finish flag in the hosts queue for alive tests.

### PR DESCRIPTION
**Pull request by jjnicola**
**What**:
If the flag to indicate that the alive test has finished has been already set at the end of the list, with these changes it is possible to add/move a host and then re allocate the flag at the end again.
**Why**:
In some cases a host can be added/moved in the alive host list.

**How**:

**Checklist**:

- [ ] Tests
- [x] [CHANGELOG](https://github.com/greenbone/gvm-libs/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

File: boreas/boreas_io.c [hunk 1]
```diff
@@ -238,6 +238,27 @@ put_host_on_queue (kb_t kb, char *addr_str)
...
+ * @brief Reallocate finish signal at the end of alive detection queue.
+ *
+ * @param main_kb  kb to use
+ */
+void
+realloc_finish_signal_on_queue(kb_t main_kb)
+{
+  int kb_item_push_str_err, pos;
+
+  pos = 1; // Append the item at the end of the queue.
```
Comment (mattmundell) [inline boreas/boreas_io.c:275]: Doesn't 1 put it at the start of the queue, with LPUSH?

File: boreas/boreas_io.c [hunk 2]
```diff
@@ -238,6 +238,27 @@ put_host_on_queue (kb_t kb, char *addr_str)
...
+  kb_item_push_str_err = kb_item_add_str_unique (main_kb,
+                                                  ALIVE_DETECTION_QUEUE,
+                                                  ALIVE_DETECTION_FINISHED,
+                                                  0, pos);
```
Comment (mattmundell) [inline boreas/boreas_io.c]:
```suggestion
                                                 ALIVE_DETECTION_QUEUE,
                                                 ALIVE_DETECTION_FINISHED,
                                                 0, pos);
```

File: boreas/boreas_io.c [hunk 3]
```diff
@@ -238,6 +238,27 @@ put_host_on_queue (kb_t kb, char *addr_str)
              __func__, addr_str);
 }
 
+/**
+ * @brief Reallocate finish signal at the end of alive detection queue.
+ *
+ * @param main_kb  kb to use
+ */
+void
+realloc_finish_signal_on_queue(kb_t main_kb)
```
Comment (mattmundell) [inline boreas/boreas_io.c]:
```suggestion
realloc_finish_signal_on_queue (kb_t main_kb)
```

File: boreas/boreas_io.h [hunk 4]
```diff
@@ -34,6 +34,9 @@ put_host_on_queue (kb_t, char *);
 void
 put_finish_signal_on_queue (void *);
 
+void
+realloc_finish_signal_on_queue(kb_t);
```
Comment (mattmundell) [inline boreas/boreas_io.h]:
```suggestion
realloc_finish_signal_on_queue (kb_t);
```

File: boreas/boreas_io.h [hunk 5]
```diff
@@ -37,6 +37,9 @@ put_finish_signal_on_queue (void *);
 void
 realloc_finish_signal_on_queue(kb_t);
 
+int
+finish_signal_on_queue(kb_t);
```
Comment (mattmundell) [inline boreas/boreas_io.h]:
```suggestion
finish_signal_on_queue (kb_t);
```

File: boreas/boreas_io.c [hunk 6]
```diff
@@ -238,13 +238,38 @@ put_host_on_queue (kb_t kb, char *addr_str)
...
 
+/**
+ * @brief Checks if the finish signal is at the end of alive detection
+ * queue.
+ *
+ * @param main_kb  kb to use
+ * @return 1 if it is already set. 0 otherwise.
+ */
+int
+finish_signal_on_queue(kb_t main_kb)
```
Comment (mattmundell) [inline boreas/boreas_io.c]:
```suggestion
finish_signal_on_queue (kb_t main_kb)
```

File: base/hosts.h [hunk 7]
```diff
@@ -180,8 +180,7 @@ gvm_host_get_addr6 (const gvm_host_t *, struct in6_addr *);
 void
 gvm_host_add_reverse_lookup (gvm_host_t *);
 
-void
-gvm_host_free (gpointer);
+void gvm_host_free (gpointer);
```
Comment (mattmundell) [inline base/hosts.h:183]:
```suggestion
void
gvm_host_free (gpointer);
```

File: boreas/boreas_io.h [hunk 8]
```diff
@@ -34,11 +34,9 @@ put_host_on_queue (kb_t, char *);
 void
 put_finish_signal_on_queue (void *);
 
-void
-realloc_finish_signal_on_queue(kb_t);
+void realloc_finish_signal_on_queue (kb_t);
```
Comment (mattmundell) [inline boreas/boreas_io.h:37]: Not sure why you're moving these down when the others are up.  Is clang-format asking for it?

**Review by mattmundell — CHANGES_REQUESTED** [review]
Not sure why this has triggered a build fail in boreas_io.c but would be nice to fix it.  It needs to include string.h.

File: boreas/boreas_io.h (hunk 8 above)
Comment (jjnicola) [reply boreas/boreas_io.h]: I have just run clang-format and change it.

File: base/hosts.h (hunk 7 above)
Comment (jjnicola) [reply base/hosts.h:183]: I tried to revert this manually, but clang-format is asking for it. So, I leave it as it is.

Comment (mattmundell) [reply base/hosts.h:183]: ok, too bad.

**Review by mattmundell — APPROVED** [review]

**Pull request merged**